casper/src/main/scala/coop/rchain/casper/Validate.scala

```diff
@@ -190,60 +190,56 @@ object Validate {
-      missingBlockStatus <- deployCountStatus.joinRight.traverse(
-                             _ => Validate.missingBlocks[F](block, dag)
-                           )
-      _ <- span.mark("before-timestamp-validation")
-      timestampStatus <- missingBlockStatus.joinRight.traverse(
-                          _ => Validate.timestamp[F](block, dag)
-                        )
-      _ <- span.mark("before-repeat-deploy-validation")
-      repeatedDeployStatus <- timestampStatus.joinRight.traverse(
-                               _ => Validate.repeatDeploy[F](block, dag, expirationThreshold, span)
-                             )
-      _ <- span.mark("before-block-number-validation")
-      blockNumberStatus <- repeatedDeployStatus.joinRight.traverse(
-                            _ => Validate.blockNumber[F](block, dag)
-                          )
-      _ <- span.mark("before-future-transaction-validation")
-      futureTransactionStatus <- blockNumberStatus.joinRight.traverse(
-                                  _ => Validate.futureTransaction[F](block)
-                                )
-      _ <- span.mark("before-transaction-expired-validation")
-      transactionExpirationStatus <- futureTransactionStatus.joinRight.traverse(
-                                      _ =>
-                                        Validate
-                                          .transactionExpiration[F](block, expirationThreshold)
-                                    )
-      _ <- span.mark("before-justification-follows-validation")
-      followsStatus <- transactionExpirationStatus.joinRight.traverse(
-                        _ => Validate.justificationFollows[F](block, genesis, dag)
-                      )
-      _ <- span.mark("before-parents-validation")
-      parentsStatus <- followsStatus.joinRight.traverse(
-                        _ => Validate.parents[F](block, genesis, dag)
-                      )
-      _ <- span.mark("before-sequence-number-validation")
-      sequenceNumberStatus <- parentsStatus.joinRight.traverse(
-                               _ => Validate.sequenceNumber[F](block, dag)
-                             )
-      _ <- span.mark("before-justification-regression-validation")
-      justificationRegressionsStatus <- sequenceNumberStatus.joinRight.traverse(
-                                         _ =>
-                                           Validate.justificationRegressions[F](block, genesis, dag)
-                                       )
-      _ <- span.mark("before-shrad-identifier-validation")
-      shardIdentifierStatus <- justificationRegressionsStatus.joinRight.traverse(
-                                _ => Validate.shardIdentifier[F](block, shardId)
-                              )
-    } yield shardIdentifierStatus.joinRight
+  ): F[Either[BlockStatus, ValidBlock]] = {
+    val getBlockSummary: EitherT[F, BlockStatus, ValidBlock] = for {
+      _ <- EitherT.liftF(span.mark("before-block-hash-validation"))
```

Comment:
The explicit lifting boilerplate looks horrible. Isn't it somehow possible to do the lifting implicitly?

Reply:
I can only think of writing an `Ops` that cats does not have and instead of doing 

```
EitherT.liftF(span.mark("before-block-hash-validation"))
```

do

```
span.mark("before-block-hash-validation").liftF
```

Reply:
but what about an implicit conversionl as @lukasz-golebiewski says. Doesn't that work?

Reply:
definiettly not a good idea to add some magical convertions in some global scope that would lift Fs to EitherT[F. good luck with figuring out compiler issues